roles/openshift_hosted/filter_plugins/filters.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+'''
+Custom filters for use in openshift_hosted
+'''
+
+
+class FilterModule(object):
+    ''' Custom ansible filters for use by openshift_hosted role'''
+
+    @staticmethod
+    def get_router_replicas(replicas=None, router_nodes=None):
+        ''' This function will return the number of replicas
+            based on the results from the defined
+            openshift.hosted.router.replicas OR
+            the query from oc_obj on openshift nodes with a selector OR
+            default to 1
+
+        '''
+        # We always use what they've specified if they've specified a value
+        if replicas is not None:
+            return replicas
+
+        if (isinstance(router_nodes, dict) and
+                'results' in router_nodes and
+                'results' in router_nodes['results'] and
+                'items' in router_nodes['results']['results']):
+
+            return len(router_nodes['results']['results'][0]['items'])
```

Comment:
@detiber, I think this is a single purpose item for the filter_plugin and tightly coupling is exactly what we want since the value is very specific. 

I see your point that if the structure changed you wouldn't receive what you thought.  That is not a good solution.  The question enters in, how do we know when it doesn't have the key that we want?  That's the difficult question.  If we run them together and test them together then we'll at least have some confidence in them.  The issue is that when the previous task is skipped the variable from the register gets written in with data that's not the same as the structure would have returned with the oc_obj query.
Maybe a better solution would be to check to see if the skip occurred and then return the data:
`if 'skipped' not in router_nodes:  return router_nodes['results']['results'][0]['items']`

This is all subject to change in the next 3.6 release as the openshift_hosted role will be completely refactored.

If we did it this way we guarantee the output to be tied with the module output.